### ISSUE TITLE
shared_pointer support in javascript

### ISSUE THREAD

**Issue by dakhnod** [swig/swig#2902]
I have seen https://github.com/swig/swig/issues/798, https://github.com/swig/swig/pull/236 and https://github.com/swig/swig/pull/1365 and find this plot a bit hard to follow.

Is there support for `std::shared_ptr`? Swig 4.2.1 doesn't seem to include `std_shared_ptr.i`, which would be really useful.

Does anyone have a hint on where I could find that?

**Comment by ojwb** [comment]
SWIG doesn't yet support `std::shared_ptr` when generating Javascript wrappers.

There have been a couple of patches submitted which you've linked to, but they were incomplete and the submitters didn't respond to requests to update their patches so they were eventually closed.

**Comment by mateusz-plociennik** [comment]
There's JavaScript Evolution fork by @mmomtchev that has it implemented for napi, see https://github.com/mmomtchev/swig/pull/42
